controllers/openstackcluster_controller.go

```diff
@@ -239,40 +223,75 @@ func (r *OpenStackClusterReconciler) reconcileNormal(ctx context.Context, log lo
 		subnetOpts.NetworkID = networkList[0].ID
 		subnetList, err := networkingService.GetSubnetsByFilter(&subnetOpts)
 		if err != nil || len(subnetList) == 0 {
-			return reconcile.Result{}, errors.Errorf("failed to find subnet: %v", err)
+			return errors.Errorf("failed to find subnet: %v", err)
 		}
 		if len(subnetList) > 1 {
-			return reconcile.Result{}, errors.Errorf("failed to find only one subnet (result: %v): %v", subnetList, err)
+			return errors.Errorf("failed to find only one subnet (result: %v): %v", subnetList, err)
 		}
 		openStackCluster.Status.Network.Subnet = &infrav1.Subnet{
 			ID:   subnetList[0].ID,
 			Name: subnetList[0].Name,
 			CIDR: subnetList[0].CIDR,
 		}
+
+		if openStackCluster.Spec.ManagedAPIServerLoadBalancer {
+			err = loadBalancerService.ReconcileLoadBalancer(clusterName, openStackCluster)
+			if err != nil {
+				return errors.Errorf("failed to reconcile load balancer: %v", err)
+			}
+		}
 	} else {
 		err := networkingService.ReconcileNetwork(clusterName, openStackCluster)
 		if err != nil {
-			return reconcile.Result{}, errors.Errorf("failed to reconcile network: %v", err)
+			return errors.Errorf("failed to reconcile network: %v", err)
 		}
 		err = networkingService.ReconcileSubnet(clusterName, openStackCluster)
 		if err != nil {
-			return reconcile.Result{}, errors.Errorf("failed to reconcile subnets: %v", err)
+			return errors.Errorf("failed to reconcile subnets: %v", err)
 		}
 		err = networkingService.ReconcileRouter(clusterName, openStackCluster)
 		if err != nil {
-			return reconcile.Result{}, errors.Errorf("failed to reconcile router: %v", err)
+			return errors.Errorf("failed to reconcile router: %v", err)
 		}
 		if openStackCluster.Spec.ManagedAPIServerLoadBalancer {
 			err = loadBalancerService.ReconcileLoadBalancer(clusterName, openStackCluster)
 			if err != nil {
-				return reconcile.Result{}, errors.Errorf("failed to reconcile load balancer: %v", err)
+				return errors.Errorf("failed to reconcile load balancer: %v", err)
 			}
 		}
 	}
 
 	err = networkingService.ReconcileSecurityGroups(clusterName, openStackCluster)
 	if err != nil {
-		return reconcile.Result{}, errors.Errorf("failed to reconcile security groups: %v", err)
+		return errors.Errorf("failed to reconcile security groups: %v", err)
+	}
+
+	return nil
+}
+
+func (r *OpenStackClusterReconciler) reconcileNormal(ctx context.Context, log logr.Logger, patchHelper *patch.Helper, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster) (ctrl.Result, error) {
```

Comment:
Can you reorder the methods please and put reconcileNormal above reconcileNetworkComponents?  (imho just a lot easier to read it "top down")

Reply:
Will do !